pkg/client/client.go

```diff
@@ -0,0 +1,195 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/aep-dev/aep-lib-go/pkg/api"
+)
+
+func Create(ctx context.Context, r *api.Resource, c *http.Client, serverUrl string, body map[string]interface{}, parameters map[string]string) (map[string]interface{}, error) {
+	suffix := ""
+	if r.CreateMethod != nil && r.CreateMethod.SupportsUserSettableCreate {
+		id, ok := body["id"]
+		if !ok {
+			return nil, fmt.Errorf("id field not found in %v", body)
```

Comment:
sometimes id fields are optional - should it still be an error here?

my vote would be to leave this validation server-side and to capture the error appropriately. We could also parse a required annotation and use that if we really wanted to, but I think we've agreed to rely on server-side validation for the most part.